### ISSUE TITLE
Synced flush deprecation message issue in v7.9.2

### ISSUE THREAD

**Issue by wasserman** [elastic/elasticsearch#65235]
**Elasticsearch version** (`bin/elasticsearch --version`):
7.9.2

**Plugins installed**: []
n/a
**JVM version** (`java -version`):
n/a
**OS version** (`uname -a` if on a Unix-like system):
n/a
**Description of the problem including expected versus actual behavior**:
On successful synced flush the deprecation warning improperly suggests `_/flush` instead of `/_flush`
The deprecation warning does not show up on failures, yet this would be useful since it would encourage the user to not to keep retrying as they are used to doing and use the preferred method.

**Steps to reproduce**:
 1. Run POST _flush/synced
 2. If shards failed, HTTP 409 - Conflict returned.  No deprecation warning.
 3. If success, HTTP 200, then the following warning:
`#! Deprecation: Synced flush is deprecated and will be removed in 8.0. Use flush at _/flush or /{index}/_flush instead.
`
**Provide logs (if relevant)**:

**Comment by dnhatn** [comment]
It seems that we not return deprecation messages unless the rest status is 2xx. Would someone from @elastic/es-core-infra take a look?

**Comment by jaymode** [comment]
@wasserman @dnhatn I took a look and could not reproduce the issue where the warning is not returned. Can you provide a reproduction?

I did find the issue with the deprecation message and will fix that shortly.

**Comment by wasserman** [comment]
@jaymode In Kibana v7.9.2 DevTools run:
`POST _flush/synced`

If failed shared HTTP 409 then no warning.
```
{
  "_shards" : {
    "total" : 736,
    "successful" : 734,
    "failed" : 2
  },
```

If successful (no failed shards) then warning:
```
#! Deprecation: Synced flush is deprecated and will be removed in 8.0. Use flush at _/flush or /{index}/_flush instead.
{
  "_shards" : {
    "total" : 736,
    "successful" : 736,
    "failed" : 0
  },
```

**Comment by rjernst** [comment]
Fixed by #66182
